src/app/admin/admin-reports/filtered-collections/filtered-collections.component.ts

```diff
@@ -0,0 +1,58 @@
+import { Component, ViewChild } from '@angular/core';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { NgbAccordion } from '@ng-bootstrap/ng-bootstrap';
+import { Observable } from 'rxjs';
+import { RestRequestMethod } from 'src/app/core/data/rest-request-method';
+import { DspaceRestService } from 'src/app/core/dspace-rest/dspace-rest.service';
+import { RawRestResponse } from 'src/app/core/dspace-rest/raw-rest-response.model';
+import { environment } from 'src/environments/environment';
+import { FiltersComponent } from '../filters-section/filters-section.component';
+import { FilteredCollections } from './filtered-collections.model';
+
+@Component({
+  selector: 'ds-report-filtered-collections',
+  templateUrl: './filtered-collections.component.html',
+  styleUrls: ['./filtered-collections.component.scss']
+})
+export class FilteredCollectionsComponent {
+
+  queryForm: FormGroup;
+  results: FilteredCollections = new FilteredCollections();
+  @ViewChild('acc') accordionComponent: NgbAccordion;
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private restService: DspaceRestService) {}
+
+  ngOnInit() {
+    this.queryForm = this.formBuilder.group({
+      filters: FiltersComponent.formGroup(this.formBuilder)
+    });
+  }
+
+  filtersFormGroup(): FormGroup {
+    return this.queryForm.get('filters') as FormGroup;
+  }
+
+  getGroup(filterId: string): string {
+    return FiltersComponent.getGroup(filterId).id;
+  }
+
+  submit() {
+    this
+      .postFilteredCollections()
+      .subscribe(
+        response => {
+          this.results.deserialize(response.payload);
+          this.accordionComponent.expand('collections');
+        }
+      );
+  }
+
+  postFilteredCollections(): Observable<RawRestResponse> {
+    let form = this.queryForm.value;
+    let scheme = environment.rest.ssl ? 'https' : 'http';
+    let urlRestApp = `${scheme}://${environment.rest.host}:${environment.rest.port}${environment.rest.nameSpace}`;
+    return this.restService.request(RestRequestMethod.POST, `${urlRestApp}/api/contentreport/filteredcollections`, form);
+  }
```

Comment:
In order not to duplicate code like this all across the app, and to be able to cache responses, please use a Dataservice for this instead. While a request like this might have been a POST(i.e. a form submit) in dspace 6, with a REST api, POSTs don't make a lot of sense in this context, as you're not creating anything. This should be a GET endpoint. And when it's a GET you can likely get away with simply extending `HALDataService<FilteredCollections>` in your DataService.

The FilteredCollection model will also need [cerialize annotations](https://github.com/weichx/cerialize) (basically put `@autoSerialize` above each property that comes from rest), and then you don't need to (de)serialize it manually anymore either. 

Same for FilteredCollections, however because collections and summary are complex types you'll likely need an `@autoserializeAs(FilteredCollection[])` for those

You'll also need to add a property for the type (I see that's already in the rest response, just not in the model in angular) and add the @typedObject annotation to the class, a static type property as well as a regular type property, as [you can see here](https://github.com/DSpace/dspace-angular/blob/143b7c3e0d541a85dc289a205bb33fe5d896a3c3/src/app/core/data/root.model.ts#L12-L21)

Reply:
We will need guidance here too. We tried at first to build some DataService, but to no avail, so we decided to use lower-level REST queries directly in the component. If we can move this code to a full-featured DataService, of course it might be better.